src/main/scala/ch/openolitor/stammdaten/StammdatenUpdateService.scala

```diff
@@ -371,23 +371,33 @@ class StammdatenUpdateService(override val sysConfig: SystemConfig) extends Even
               stammdatenWriteRepository.updateEntityFully[DepotlieferungAbo, AboId](copy)
             }
           case abo: Abo =>
-            // wechsel
-            val aboNeu = copyTo[HauptAbo, DepotlieferungAbo](
-              abo,
-              "depotId" -> va.depotId,
-              "depotName" -> depot.name,
-              "vertriebId" -> va.vertriebId,
-              "vertriebsartId" -> update.vertriebsartIdNeu
-            )
-            abo match {
-              case abo: HeimlieferungAbo =>
-                stammdatenWriteRepository.deleteEntity[HeimlieferungAbo, AboId](abo.id)
-                stammdatenWriteRepository.deleteEntity[Tourlieferung, AboId](abo.id)
-              case abo: PostlieferungAbo => stammdatenWriteRepository.deleteEntity[PostlieferungAbo, AboId](abo.id)
-              case _                     =>
-            }
+            stammdatenWriteRepository.getById(vertriebMapping, va.vertriebId) map { vertrieb =>
+              // wechsel
+              val aboNeu = copyTo[HauptAbo, DepotlieferungAbo](
+                abo,
+                "depotId" -> va.depotId,
+                "depotName" -> depot.name,
+                "vertriebId" -> va.vertriebId,
+                "vertriebsartId" -> update.vertriebsartIdNeu,
+                "vertriebBeschrieb" -> vertrieb.beschrieb
+              )
+              abo match {
+                case abo: HeimlieferungAbo =>
+                  stammdatenWriteRepository.deleteEntity[HeimlieferungAbo, AboId](abo.id)
+                  stammdatenWriteRepository.deleteEntity[Tourlieferung, AboId](abo.id)
+                case abo: PostlieferungAbo => stammdatenWriteRepository.deleteEntity[PostlieferungAbo, AboId](abo.id)
```

Comment:
For me this looks strange. On HeimlieferungAbo also delete Tourlieferung?!

Reply:
In case an abo changes the vertrieb, the type can change. So, it needs to be deleted from heim or tour and, for example, added to depot.